### DISCUSSION TITLE
OpenRBG cannot target the light zones that G-H can

### DISCUSSION THREAD

**Discussion by iamtimsson** [seerge/g-helper#747] [Q&A] [answered]
Being both open source projects, I am hoping to do the work of finding out how the light zones are targeted in g-h so that I may apply that targetting to openrgb.

I hope you can see that the proposed solution and its journey is good work in the [microphone issue](https://github.com/seerge/g-helper/issues/744#issue-1794780858) that was posted. I get very frustrated on my own - someone's wisdom quells the pain - I could use a hand. Thanks! I am doing my similar discussion searches now while answering what I can along the way. Will post details as I go.

It seems the keyboard zones are controlled by [usb?](https://github.com/seerge/g-helper/discussions/344#discussioncomment-5849339).
It seems that g-h uses the [registry](https://github.com/seerge/g-helper/discussions/342#discussioncomment-5952920) to set charging limits for battery?
It seems optimization service [overwrites hotkeys](https://github.com/seerge/g-helper/discussions/342#discussioncomment-5957193)?
YO? It seems the backlight is controlled in g-h through [windows api](https://github.com/seerge/g-helper/discussions/461#discussioncomment-6009632)?? I don't know what to do with this but I know what those words mean.
It seems the rog ally backlight is [controlled without modification](https://github.com/seerge/g-helper/discussions/461#discussioncomment-6203177) by g-h so there must be something openrgb is not doing that g-h is. :hmm:
It seems not long ago, [toggles were implemented](https://github.com/seerge/g-helper/discussions/303#discussioncomment-5813696) for the backlight zones. 
SOLUTION? It seems that if I [search for asus usb devices](https://github.com/seerge/g-helper/discussions/376#discussioncomment-5893286), we can point to them? 

A place my ability has always been stunted is having the knowledge to apply my wisdom in a sense. I know that somewhere in this repo there is a ui file that has an extras button that has in its ui a button or whatnot that executes a function that pushes the backlight data somewhere. I cannot figure out how to sniff this out. Helping me learn this is teaching a skill not giving a handout. Much love.

**Comment by seerge** [comment]
@iamtimsson hello. All backlight controls are inside [AsusUSB](https://github.com/seerge/g-helper/blob/e0cca33ab9efa2158063acb326505efe75f48060/app/AsusUSB.cs#L340) G-Helper doesn't target zones, it sets backlight for a whole keyboard all together (but it can just turn certain "backlight" parts on or off)

**Comment by iamtimsson** [comment]
Damn it thanks seerge. I just don't know how to interpret what I am looking at I don't have the mind for it.
I can sit there any apply my self and just get frustrated and no where else.
All I know is that g-h has no problems setting the lighting on the scar 17 g733zw but openrpg doesn't set the 10key.
I guess I'll go bug them you've done a lot.

**Comment by iamtimsson** [comment, marked as answer]
I have [posted](https://github.com/fjardon/OpenRGB/issues/5#issue-1795449326) at OpenRGB but that seems to be a dead repo so it is [here](https://gitlab.com/CalcProgrammer1/OpenRGB/-/issues/3472)
